### PR TITLE
User role assignment

### PR DESCRIPTION
**Pull request by TobySalusky**

### CODE REVIEW

File: common/user.js [hunk 1]
```diff
@@ -460,6 +469,71 @@ class User {
...
+      noRoles: true,
+      sortField: 'userId',
+      sortDescending: false
+    };
+    if (req.body.filter) { query.filter = req.body.filter; }
+    query.searchFields = ['userId', 'userName'];
+
+    Promise.all([
+      User.searchUsers(query),
+      User.numberOfUsers()
```
Comment (awick) [inline common/user.js:506]: i dont see the counts in the UI, do we need the numberOfUsers call and data?

Comment (TobySalusky) [reply common/user.js:506]: Oops, looks like I forgot to remove that from back when I was still planning on using pagination.

File: common/user.js [hunk 2]
```diff
@@ -460,6 +469,71 @@ class User {
...
+   * @typedef ArkimeUserInfo
+   * @type {object}
+   * @param {string} userId - The ID of the user.
+   * @param {string} userName - The name of the user (to be displayed in the UI).
+   * @param {boolean | undefined} hasRole - whether the user has the requested role
+   *            (only if a role was provided & the requester is a roleAssigner for it)
+   */
+
+  /**
+   * POST - /api/users/list
```
[the remark this answers is not in the document]
Comment (TobySalusky) [reply common/user.js:484]: /api/users/min sounds good to me. I think combining the routes with a flag would make the permissions ugly and cause  each use-case to have 1 or more un-usable/irrelevant parameters.

File: common/user.js [hunk 3]
```diff
@@ -434,9 +441,11 @@ class User {
     if (req.body.filter) {
       query.filter = req.body.filter;
     }
+    query.noRoles = !!req.body.noRoles;
```
[the remark this answers is not in the document]
Comment (TobySalusky) [reply common/user.js]: Oh, that's from before I had made the new route, I've now fixed it.

**Pull request merged**